ptvsd/wrapper.py

```diff
@@ -739,15 +739,37 @@ def on_variables(self, request, args):
 
         variables = []
         for xvar in xvars:
+            var_name = unquote(xvar['name'])
             var = {
-                'name': unquote(xvar['name']),
+                'name': var_name,
                 'type': unquote(xvar['type']),
                 'value': unquote(xvar['value']),
             }
             if bool(xvar['isContainer']):
-                pyd_child = pyd_var + (var['name'],)
+                pyd_child = pyd_var + (var_name,)
                 var['variablesReference'] = self.var_map.to_vscode(
                     pyd_child, autogen=True)
+
+            if len(pyd_var) > 3:
+                # This means the current variable has a parent i.e, it is not a
+                # FRAME variable. These require evaluateName to work in VS 
+                # watch window
+                eval_name = pyd_var[3]
+                for s in pyd_var[4:]:
+                    try:
+                        # In the result XML from pydevd list indexes appear 
+                        # as names. If the name is a number then it is a index.
+                        eval_name += '[' + str(int(s)) + ']'
```

Comment:
Can you check what it does for dicts?

Reply:
This will not work for dicts. The `name` returned by pydevd is in a strange format. For example, consider this dict:
```
c= ('green', 'blue')
dict1 = {
    'apple': 'Red',
    0 : 'Zero',
    c: 'colors',
    }
```
This is how the `name`, `type`, and `value` appear to us:

| Name                             | Type  | Value    |
|----------------------------------|-------|----------|
| 'apple' (2200118833872)          | str   | 'Red'    |
|('green', 'blue') (2200118705544) | str   | 'colors' |
| 0 (140728987558560)              | str   | 'Zero'   |
| `__len__`                          | int   | 3        |